proxy/src/http/source.rs

```diff
@@ -390,53 +394,35 @@ mod handler {
     pub async fn revisions(
         peer: Arc<Mutex<coco::PeerApi>>,
         project_urn: String,
+        owner: coco::User,
     ) -> Result<impl Reply, Rejection> {
-        let peer = peer.lock().await;
         let urn = project_urn.parse().map_err(Error::from)?;
-        let (branches, tags) = coco::with_browser(&peer, &urn, |browser| {
-            Ok((coco::branches(browser)?, coco::tags(browser)?))
-        })?;
-
-        // TODO(rudolfs): the order of the returned peers/revisions determines the default peer in
-        // the repository selector in the UI. Make sure the list always returns the default peer
-        // first.
-        let revs = [
-            (
-                "cloudhead",
-                "rad:git:hwd1yre85ddm5ruz4kgqppdtdgqgqr4wjy3fmskgebhpzwcxshei7d4ouwe",
-            ),
-            (
-                "rudolfs",
-                "rad:git:hwd1yrereyss6pihzu3f3k4783boykpwr1uzdn3cwugmmxwrpsay5ycyuro",
-            ),
-            (
-                "xla",
-                "rad:git:hwd1yreyu554sa1zgx4fxciwju1pk77uka84nrz5fu64at9zxuc8f698xmc",
-            ),
-        ]
-        .iter()
-        .map(|(fake_handle, fake_peer_urn)| super::Revision {
-            branches: branches.clone(),
-            tags: tags.clone(),
-            identity: identity::Identity {
-                // TODO(finto): Get the right URN
-                id: fake_peer_urn
-                    .parse()
-                    .expect("failed to parse hardcoded URN"),
-                metadata: identity::Metadata {
-                    handle: (*fake_handle).to_string(),
-                },
-                avatar_fallback: avatar::Avatar::from(fake_handle, avatar::Usage::Identity),
-                registered: None,
-                shareable_entity_identifier: identity::SharedIdentifier {
-                    handle: (*fake_handle).to_string(),
-                    urn: fake_peer_urn
-                        .parse()
-                        .expect("failed to parse hardcoded URN"),
-                },
-            },
-        })
-        .collect::<Vec<super::Revision>>();
+        let peer = &*peer.lock().await;
+        let remotes = coco::remotes(peer, &owner, &urn)?;
+
+        let revs = remotes
+            .into_iter()
+            .map(|remote| {
+                let urn = remote.user.urn();
+                let handle = remote.user.name().to_string();
+                super::Revision {
+                    branches: remote.branches,
+                    tags: remote.tags,
+                    identity: identity::Identity {
+                        id: urn.clone(),
+                        metadata: identity::Metadata {
+                            handle: handle.clone(),
+                        },
+                        avatar_fallback: avatar::Avatar::from(
```

Comment:
Note to self: add `Avatar::from_urn`, the string conversion is making me nervous.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/636/commits/b2536a4d58f78b8368eb9c521e5efe8932eb2787